wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -1565,9 +1565,11 @@ impl crate::Adapter<super::Api> for super::Adapter {
             }
         };
 
+        let mut uniques = std::collections::HashSet::new();
```

Comment:
This is a kinda weird pattern, could you add a comment that explains what's going on.